### PR TITLE
add upgrade

### PR DESCRIPTION
**Pull request by hparfr**
And refactore determine_db

### CODE REVIEW

File: ak.py [hunk 1]
```diff
@@ -19,6 +19,45 @@
...
+    dbParams = {
+        "db_host": "PGHOST",
+        "db_port": "PGPORT",
+        "db_name": "PGDATABASE",
+        "db_user": "PGUSER",
+        "db_password": "PGPASSWORD"
+    }
+    db = cli.SwitchAttr(["d"], str, help="Database")
+
+    def determine_db(self):
```
Comment (rvalyi) [inline ak.py]: determine looks rather "frenchouillard", o at least it's not the right meaning in English I think. May be get_db or find_db...

**Comment by rvalyi** [comment]
Hello, I'm not really convinced: is it worth 27 lines of code just of a saving the user to type:
./bin/upgrade_openerp -d db ?
looks a bit like a leaky abstraction for me. On one side it looks totally ok to wrapp ak run or ak build because we run them very often, but do it so often, but is it justified for the upgrade script? I don't know, just asking in fact. Do you see a situation where customers or power users on customer side woul typically benefit from having such simpler commands available?

**Comment by rvalyi** [comment]
@legalsylvain in general, I think critical migrations will rather happen on the production server side than on the dev workstation with Voodoo: the reasoning is that often a database is 300 Mo or more and you also want to minimized downtime. Finally

**Comment by legalsylvain** [comment]
@rvalyi, You're right, the use of ak upgrade is theoritical. Not a clear PoV on that PR. @sebastienbeau , time to githubing ... ;-)

File: ak.py (hunk 1 above)
Comment (hparfr) [inline ak.py]: ok, may be set_db instead because, it set db credentials in the environment variables

**Comment by hparfr** [comment]
@rvalyi I agree with the less code there is; the better it is to maintain.

In the source code, it's less than 10 lines of python code.

From an UX perspective, the main benefits are : 
- consistancy : same behavior than other ak * commands
- discoverabilty : autodocumented with ak --help , easy to guess, predictable from this kind of tool
- efficiency :  no need to specify the database name (it's taken from .cfg file).

From my personal usage, I run it on a daily basis when I work on a heavy project.

File: ak.py [hunk 2]
```diff
@@ -89,6 +128,19 @@ def main(self, *args):
...
 
+@Ak.subcommand("upgrade")
+class AkUpgrade(cli.Application, DbTools):
+    """Upgrade odoo."""
+
+    def main(self, *args):
+        db = self.determine_db()
+        params = []
+        params += ['-d', db]
+        command = 'bin/upgrade_openerp'
```
Comment (sebastienbeau) [inline ak.py]: Hi maybe I miss something but when you run bin/upgrade_openerp the script "upgrade_openerp" will already read the config file. We only have to pass a new db if we want to force manually a specific db

Comment (hparfr) [inline ak.py]: ah ah ah ! Yes ! 
:-1: for me to not reading the source.

**Comment by hparfr** [comment]
> On one side it looks totally ok to wrapp ak run or ak build because we run them very often, but do it so often, but is it justified for the upgrade script? 

I think there is two type of commands : 
-  daily usage : should be shortcuts commands :  ak run, ak db
-  rare usage : loading a database, building the project with forzen.cfg, freezing a project, doing a git diff on submodules, ...

File: ak.py (hunk 2 above)
Comment (sebastienbeau) [inline ak.py]: It's the same for start_openerp and python_openerp

**Pull request merged**